File: WebLab3/src/main/java/com/arslanka/backend/models/views/Rectangle.java
package com.arslanka.backend.models.views;

public class Rectangle implements Shape {
    private final double xMultiplier;
    private final double yMultiplier;

    public Rectangle(Distance xRad, Distance yRad) {
        if (xRad == Distance.HALF_R) {
            xMultiplier = 2.0d;
        } else {
            xMultiplier = 1.0d;
        }
        if (yRad == Distance.HALF_R) {
            yMultiplier = 2.0d;
        } else {
            yMultiplier = 1.0d;
        }
    }

    @Override
    public Boolean isHit( Double radius, Double x, Double y) {
        x = Math.abs(x);
        y = Math.abs(y);
        return (x * xMultiplier) <= radius && (y * yMultiplier) <= radius;
    }
}

File: WebLab2/src/main/kotlin/com/arslanka/weblab2/models/view/Rectangle.kt
package com.arslanka.weblab2.models.view

import java.math.BigDecimal

class Rectangle(xDistance: Distance, yDistance: Distance) : Shape {
    private var xMultiplier = BigDecimal.ZERO
    private var yMultiplier = BigDecimal.ZERO

    init {
        xMultiplier = if (xDistance === Distance.HALF_R) BigDecimal.valueOf(2) else BigDecimal.ONE
        yMultiplier = if (yDistance === Distance.HALF_R) BigDecimal.valueOf(2) else BigDecimal.ONE
    }

    override fun isHit(radius: BigDecimal, x: BigDecimal, y: BigDecimal): Boolean {
        return (x.abs() * xMultiplier) <= radius && (y.abs() * yMultiplier <= radius)
    }
}